batcher.go

```diff
@@ -33,46 +29,31 @@ type Batcher interface {
 	ID() string
 	// Category defines if a Forward or Return
 	Category() string
-}
-
-// BatchError is an Error that describes batch validation issues
-type BatchError struct {
-	BatchNumber int
-	FieldName   string
-	Msg         string
-}
-
-func (e *BatchError) Error() string {
-	return fmt.Sprintf("BatchNumber %d %s %s", e.BatchNumber, e.FieldName, e.Msg)
+	Error(string, error, ...interface{}) error
 }
 
 // Errors specific to parsing a Batch container
 var (
-	// generic messages
-	msgBatchHeaderControlEquality     = "header %v is not equal to control %v"
-	msgBatchCalculatedControlEquality = "calculated %v is out-of-balance with batch control %v"
-	msgBatchAscending                 = "%v is less than last %v. Must be in ascending order"
 	// specific messages for error
+	msgBatchCardTransactionType   = "Card Transaction Type %v is invalid"
```

Comment:
These are just rearranged to help me see the patterns across the various types of errors